staging/src/k8s.io/apiextensions-apiserver/test/integration/fieldselector_test.go

```diff
@@ -593,9 +592,7 @@ func TestFieldSelectorOpenAPI(t *testing.T) {
 
 func TestFieldSelectorDropFields(t *testing.T) {
 	_, ctx := ktesting.NewTestContext(t)
-	featuregatetesting.SetFeatureGateEmulationVersionDuringTest(t, utilfeature.DefaultFeatureGate, version.MustParse("1.31"))
-	featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, apiextensionsfeatures.CustomResourceFieldSelectors, false)
-	tearDown, apiExtensionClient, _, err := fixtures.StartDefaultServerWithClients(t)
+	tearDown, apiExtensionClient, _, err := fixtures.StartDefaultServerWithClients(t, "--emulated-version=1.31", "--feature-gates=CustomResourceFieldSelectors=false")
```

Comment:
re feature gates via flags: Is this only necessary for changing feature gates while emulating a previous version or are we eventually going to use this format for all feature gate changes in integration tests?

Reply:
Ideally I would like all feature gate changes go through the flag.  But that is not feasible when testing multiple components like with scheduler, so `SetFeatureGateDuringTest` will stay. So using the flags is really necessary for changing feature gates while emulating a previous version. Other cases, `SetFeatureGateDuringTest` is probably fine.